### PR TITLE
Fix logClientIn for non-existing users

### LINKED ISSUE #26123: Webdav wrong error PasswordLoginForbidden when specifying non-existing user
Steps:

1. Pick a user that doesn't exist, for example "unexist"
1. Run this:
```
 % curl -X PROPFIND http://unexist:x@localhost/owncloud/remote.php/webdav/
<?xml version="1.0" encoding="utf-8"?>
<d:error xmlns:d="DAV:" xmlns:s="http://sabredav.org/ns" xmlns:o="http://owncloud.org/ns">
  <s:exception>OCA\DAV\Connector\Sabre\Exception\PasswordLoginForbidden</s:exception>
  <s:message/>
  <o:hint xmlns:o="o:">password login forbidden</o:hint>
</d:error>
```

### Expected result
A simple 401 without extra exception

### Actual result
PasswordLoginForbidden.

### Versions
master / 9.2 pre-alpha

This exception is only intended to be used for when two factor is enabled and when a client must use tokens instead of passwords.

9.1 is likely affected too.

@DeepDiver1975 @SergioBertolinSG

### PR DESCRIPTION
**Pull request by PVince81**
## Description

This prevents throwing PasswordLoginForbidden for non-existing users.
## Related Issue

Fixes https://github.com/owncloud/core/issues/26123
## Motivation and Context

The check for two factor enforcement would return true for non-existing
users. This fix makes it return false in order to be able to perform
the regular login which will then fail and return false.
## How Has This Been Tested?

See original steps.
Also enable two factor auth and check that for existing users the exception is still PasswordLoginForbidden in the DAV response.
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
## Backports
- [x] stable9.1

### CODE REVIEW

**Comment by PVince81** [comment]
Please review @DeepDiver1975 @jvillafanez @VicDeo

**Review by DeepDiver1975 — APPROVED** [review]

File: tests/lib/User/SessionTest.php [hunk 1]
```diff
@@ -384,6 +384,32 @@ public function testLogClientInNoTokenPasswordWith2fa() {
 		$userSession->logClientIn('john', 'doe', $request);
 	}
 
+	public function testLogClientInUnexist() {
+		$manager = $this->getMockBuilder('\OC\User\Manager')
```
Comment (PVince81) [inline tests/lib/User/SessionTest.php]: okay... thing is I just copy-pasted the previous test and adjusted it.

Noted for the future, there's a lot to adjust.

**Comment by DeepDiver1975** [comment]
:+1:

**Comment by DeepDiver1975** [comment]
backporting ....

**Pull request merged**